### PR TITLE
feat(allow-whitelist-keys): allow whitelist keys in metadata

### PR DESCRIPTION
**Pull request by idonava**
Filtering resource metadata by allows keys, Only branches that contain keys from keys_to_allow will be sent to epsagon.

### CODE REVIEW

File: epsagon/trace.py [hunk 1]
```diff
@@ -126,6 +128,7 @@ def initialize(
         :param url_patterns_to_ignore: URL patterns to ignore in HTTP data
          collection.
         :param keys_to_ignore: List of keys to ignore while extracting metadata.
```
Comment (razlo) [inline epsagon/trace.py]: which one is stronger?

File: epsagon/trace.py [hunk 2]
```diff
@@ -476,6 +483,14 @@ def __init__(
                 )
         else:
             self.keys_to_ignore = []
+        if keys_to_allow:
+            self.keys_to_allow = [self._strip_key(x) for x in keys_to_allow]
+            if self.debug:
+                print(
```
Comment (razlo) [inline epsagon/trace.py]: don't we have debug_log function here?
I think we added one here recently

File: epsagon/trace.py [hunk 3]
```diff
@@ -476,6 +483,14 @@ def __init__(
                 )
         else:
             self.keys_to_ignore = []
+        if keys_to_allow:
+            self.keys_to_allow = [self._strip_key(x) for x in keys_to_allow]
```
Comment (razlo) [inline epsagon/trace.py]: ```for key in...```

File: epsagon/trace.py [hunk 4]
```diff
@@ -955,6 +991,9 @@ def _send_traces(self):
 
         # Remove ignored keys.
```
Comment (razlo) [inline epsagon/trace.py]: update this comment

File: epsagon/trace.py [hunk 5]
```diff
@@ -887,6 +902,27 @@ def remove_ignored_keys(self, input_dict):
                     copied_dict[key] = self.remove_ignored_keys(value)
         return copied_dict
 
+    def get_dict_with_allow_keys(self, input_dict):
```
Comment (razlo) [inline epsagon/trace.py]: maybe just get_allowed_keys?

File: epsagon/trace.py [hunk 6]
```diff
@@ -887,6 +902,27 @@ def remove_ignored_keys(self, input_dict):
...
+        """
+        Keeping only the branches which contains allowed keys.
+        Any branch that doesn't contain allowed key will be deleted.
+        :param input_dict: Input dict to remove branches without allowed keys.
+        :return: a dict that contains branches which contains allowed keys
+        """
+        copied_dict = input_dict.copy()
+        for key in input_dict:
+            is_key_in_whitelist = self._strip_key(key) in self.keys_to_allow
```
Comment (razlo) [inline epsagon/trace.py]: this variable is redundant
you can just:
```
if self._strip_key(key) not in self.keys_to_allow
```

File: README.md [hunk 7]
```diff
@@ -227,6 +227,21 @@ epsagon.init(
     keys_to_ignore=['Request Data', 'Status_Code']
 )
 ```
+
+### Allow keys
+You can allow data to be sent to epsagon by filtering specific keys in initialization.
+￿￿Only branches which contains allow keys will be sent.
```
[the remark this answers is not in the document]
Comment (idonava) [reply README.md]: fixed

File: epsagon/trace.py (hunk 1 above)
Comment (idonava) [reply epsagon/trace.py]: Ignored keys. added to readme and switched running order of `remove_ignored_keys` and `get_dict_with_allow_keys` functions

File: epsagon/trace.py (hunk 2 above)
Comment (idonava) [reply epsagon/trace.py]: No

File: epsagon/trace.py (hunk 3 above)
Comment (idonava) [reply epsagon/trace.py]: fixed

File: epsagon/trace.py (hunk 5 above)
Comment (idonava) [reply epsagon/trace.py]: but it's a dictionary that contains allowed keys, and not just keys.

File: epsagon/trace.py (hunk 6 above)
Comment (idonava) [reply epsagon/trace.py]: I saw that, But I think the function is more readable and clear in this way.

File: epsagon/trace.py (hunk 4 above)
Comment (idonava) [reply epsagon/trace.py]: fixed

File: epsagon/utils.py [hunk 8]
```diff
@@ -137,6 +139,10 @@ def init(
     if ignored_keys:
         ignored_keys = ignored_keys.split(',')
 
+    allowed_keys = os.getenv('EPSAGON_ALLOWED_KEYS')
```
Comment (razlo) [inline epsagon/utils.py]: I added a section documenting env variables in the README file, please add this one too

File: README.md [hunk 9]
```diff
@@ -227,6 +228,22 @@ epsagon.init(
     keys_to_ignore=['Request Data', 'Status_Code']
 )
 ```
+
+### Allowed keys
+You can allow data to be sent to epsagon by filtering specific keys in initialization.
+Only keys included in this list will be sent to epsagon.
+Good to know - keys_to_ignore is stronger then keys_to_allow if you use both.
```
Comment (razlo) [inline README.md]: NOTE - Keys found in keys_to_ignore override this setting

Comment (idonava) [reply README.md]: fixed

File: epsagon/utils.py (hunk 8 above)
Comment (idonava) [reply epsagon/utils.py]: added

**Pull request merged**